src/remove-task.test.js

```diff
@@ -0,0 +1,22 @@
+// import { deleteTodoList } from './remove-task.js';
+const jsdom = require('jsdom');
+
+const { JSDOM } = jsdom;
+
+const dom = new JSDOM(`<!DOCTYPE html><ul class="d-flex todo-lists"></ul>`); // eslint-disable-line
+const { document } = dom.window;
+
+test('remove form the dom', () => {
+  document.body.innerHTML = `
+    <li class="list-container d-flex">
+      <input type="checkbox" class="checkbox">
+      <p class="items show"   contenteditable="true">Hello Todo's</p>
+      <i class="fas fa-ellipsis-v  dots"></i>
+      <i class="far fa-trash-alt show delete"></i>
+    </li>
+    `;
+  const element = document.querySelector('.list-container');
+  expect(element.children[2].classList.contains('show')).toBe(false);
+  expect(element.children[3].classList.contains('show')).toBe(true);
+  // expect(element.children[4].classList.contains('show')).toBe(false);
+});
```

Comment:
- [ ] You've done well implementing this test. However, the particular required change has not been implemented. Maybe the required change in the previous review was not very explicit. The previous reviewer said, "the `deleteTask()` function is never actually tested". What this referred to was this function in your `src/app.js` file:

```JavaScript
function deleteTask(idx) {
  const storage = JSON.parse(localStorage.getItem('todos'));
  const savedata = storage.filter((value, index) => index !== idx);
  savedata.forEach((obj, index) => {
    obj.index = index + 1;
  });
  localStorage.setItem('todos', JSON.stringify(savedata));
  window.location.reload();
}
```

Please write a unit test for the above function just as the previous reviewer recommended. Remember, in removing a task we are testing two parts, removing a list element from the UI, which you've done already 👍🏼, and deleting an item from the tasks array in the localStorage, which you are yet to implement.

Reply:
Dear @Lameck1  

   Thank you for your honest review, I think you need to check the Requirement, we were not asked to write a function for the local storage, we were only asked to write a function for Add and Delete which we did, the first reviewer want us to put them in a separate file, coming up with yours looks like we are trailing another direction, my honest opinion though.   


Thank you
cheers happy coding 👍

Reply:
Hello @zemola,

Since mine is a re-review, I am only supposed to check whether the requested changes were implemented. In this case, one required change for testing the `deleteTask()` function is not implemented. And this is what I am trying to clarify. I have even gone a step further to show which function you are supposed to write tests for.

As for the writing of functions, you've misunderstood my statement. I know this task is for testing your existing functions and NOT writing new ones. I am not coming up with anything new and there is nothing like a function for localStorage anywhere in this task. Please re-read my statement to understand it better. I am simply trying to explain to you what deleting a task entails.

Quoted below is the part of the requirement you've advised me to check:

> - Create a test file (`[..].test.js`) for a file containing the add item and delete item functions that you want to test.
> - Take a look at each of these functions. Are they pure functions? If the answer is "yes" then writing tests for them should be straightforward. Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks:
>     - Mock a storage object to "imitate" localStorage operations (only if you are using `jest < 24.x`, in newer versions localStorage is mocked automatically).
>     - Mock HTML to test if add/delete functions add or remove exactly one `<li>` element to/from the list in the DOM.

In the last two bullet points, it is quite evident that your test should assert that an item is removed from the localStorage and also one `<li>` is removed from the DOM.

As I said, you've taken care testing the removal of an `<li>` from the DOM, what remains is testing for removal of an item from the localStorage.

Reply:
I can clarify these concerns over a zoom call if need be.

Reply:
@Lameck1,

Thank you for taking your time to explain, we have also taken our time to make the changes as requested.